src/main/java/net/mcreator/ui/dialogs/NewModElementDialog.java

```diff
@@ -49,8 +49,10 @@ public static void showNameDialog(MCreator mcreator, ModElementType type) {
 						L10N.t("dialog.new_modelement.title_window", type.getReadableName()),
 						TiledImageCache.getModTypeIcon(type), new OptionPaneValidatior() {
 							@Override public Validator.ValidationResult validate(JComponent component) {
-								regName.setText(L10N.t("dialog.new_modelement.registry_name", RegistryNameFixer
-										.fromCamelCase(((VTextField) component).getText())));
+								String regNameString = RegistryNameFixer.fromCamelCase(((VTextField) component).getText());
+								regName.setText(L10N.t("dialog.new_modelement.registry_name", regNameString == null ?

```

Comment:
You should also check for empty string here, not just null

Reply:
:heavy_check_mark: Fixed and tested,
Works as expected.